internal/sddc/sddc_network_subresource.go

```diff
@@ -0,0 +1,204 @@
+/*
+ *  Copyright 2023 VMware, Inc.
+ *    SPDX-License-Identifier: MPL-2.0
+ */
+
+package sddc
+
+import (
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/validation"
+	utils "github.com/vmware/terraform-provider-vcf/internal/resource_utils"
+	"github.com/vmware/vcf-sdk-go/models"
+)
+
+// NetworkSpecNetworkType valid network types enumeration.
+var NetworkSpecNetworkType = []string{"VSAN", "VMOTION", "PUBLIC", "MANAGEMENT", "NSX_VTEP", "HOST_MANAGEMENT",
```

Comment:
most of the values here are obsolete.. currently the allowed values are VSAN, VMOTION, MANAGEMENT and VM_MANAGEMENT + any custom network type